front/src/modules/timesStops/hooks/useOutputTableData.ts

```diff
@@ -37,19 +41,31 @@ function useOutputTableData(
   path?: PathfindingResultSuccess
 ): TimeStopsRow[] {
   const { t } = useTranslation('timesStops');
+  const { getTrackSectionsByIds } = useScenarioContext();
 
   const scheduleByAt: Record<string, ScheduleEntry> = keyBy(selectedTrainSchedule.schedule, 'at');
-  const suggestedOperationalPoints: SuggestedOP[] = useMemo(
-    () =>
-      path?.length
-        ? formatSuggestedOperationalPoints(
-            pathProperties.operationalPoints,
-            pathProperties.geometry,
-            path.length
-          )
-        : [],
-    [pathProperties.operationalPoints, pathProperties.geometry]
-  );
+  const [suggestedOperationalPoints, setSuggestedOperationalPoints] = useState<SuggestedOP[]>([]);
+
+  useEffect(() => {
+    const fetchSuggestedOperationalPoints = async () => {
+      if (path?.length) {
+        const formattedPoints = formatSuggestedOperationalPoints(
+          pathProperties.operationalPoints,
+          pathProperties.geometry,
+          path.length
+        );
+        const formattedPointsWithTrackNames = await fetchTrackNamesForSuggestedOperationalPoints(
+          formattedPoints,
+          getTrackSectionsByIds
+        );
+        setSuggestedOperationalPoints(formattedPointsWithTrackNames);
+      } else {
+        setSuggestedOperationalPoints([]);
+      }
+    };
+
+    fetchSuggestedOperationalPoints();
+  }, [pathProperties.operationalPoints, pathProperties.geometry, getTrackSectionsByIds]);
```

Comment:
Can't we keep the `useMemo` here and avoid the state ? Or the component won't rerender if we don't ?

Reply:
Yep you're right